index.js

```diff
@@ -31,14 +34,7 @@ async function getStream(inputStream, options) {
 			reject(error);
```

Comment:
I don't think we need the promise constructor now (`new Promise`).

Reply:
I tried to remove the use on the Promise, but the problem is this callback based operation:
```js
stream.on('data', () => {
	if (stream.getBufferedLength() > maxBuffer) {
		rejectPromise(new MaxBufferError());
	}
});
```
Trying to change this I came up with the following, which emits an stream error when maxBuffer is exceeded (then the pipeline throws):
```js
	stream.on('data', () => {
		if (stream.getBufferedLength() > maxBuffer) {
			stream.emit('error', new MaxBufferError());
		}
	});

	try {
		await streamPipelinePromisified(inputStream, stream);
	} catch (error) {
		if (stream.getBufferedLength() <= BufferConstants.MAX_LENGTH) {
			error.bufferedData = stream.getBufferedValue();
		}

		throw error;
	}
```
However I'm not sure if it's ok emitting an error like this - I didn't found an entry on `.emit` on the stream docs, which could mean it's for internal use, and on [API for stream implementers](https://nodejs.org/api/stream.html#stream_api_for_stream_implementers) it says:
> Avoid overriding public methods such as write(), end(), cork(), uncork(), read() and destroy(), or emitting internal events such as 'error', 'data', 'end', 'finish' and 'close' through .emit()

Another try I did to remove the promise was using `for await`, however wasn't successful https://github.com/sindresorhus/get-stream/issues/21#issuecomment-671107656  (I didn't investigate in depth, so I may have done a mistake on my coding)

Reply:
Can you not use `util.promisify` on the `stream.on('data')` call?

Reply:
I don't know, I never used `util.promisify` for wrapping an event on an EventEmitter - it seems a little odd because we could have multiple `data` events, and then we would have multiple promises one after another to be awaited (and we have the problem that at some point 'data' wouldn't be emitted anymore, so we would have to handle that too, I think)

Reply:
Ok. Let's skip this until we do the for-await thing.